scripts/release/release-notes-onprem.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+
+# Usage: release-notes-onprem.sh -v [version] -t [tag/sha] -r 1
+# What will this do?
+#
+# This script helps retrieve tickets that are related to the release by:
+# 1. Find relevant tickets based on commit messages
+# 2. Format the release notes 
+# 3. Create PR with release notes
+#
+
+while [[ "$#" -gt 0 ]]
+  do
+    case $1 in
+      # If no dev tag is supplied, it will assume the version that is currently in the dev environment is being deployed
+      -t|--tag) DEV_TAG="$2"; shift;;
+      # If jira is enabled, then it will create release version, and attach tickets to it
+      -r|--release) IS_RELEASE="$2"; shift;;
+      # This is required
+      -v|--version) RELEASE_VERSION="$2"; shift;;
+    esac
+    shift
+done
+
+####################
+# Configurtation
+####################
+DOCS_ROOT_DIT=$(pwd)
+PACTFLOW_APPLICATION_DIR=$(pwd)/scripts/release/.pactflow-application
+ONPREM_PROD_IMAGE=quay.io/pactflow/enterprise:latest
+JIRA_PROJECT_ID=17612
+JIRA_URL=https://smartbear.atlassian.net
+JIRA_USER=${JIRA_AUTH%%@*}
+
+####################
+# Validation
+####################
+if [ -z ${RELEASE_VERSION} ]; then 
+  echo "Version has not be provided."
+  exit 128
+fi
+
+if [ -z ${JIRA_AUTH} ]; then 
+  echo "JIRA_AUTH not set, please set it with by exporting JIRA_AUTH=email:token, you can create a token here: https://id.atlassian.com/manage-profile/security/api-tokens"
+  exit 128
+fi
+
+
+####################
+# Checkout Application
+####################
+echo "Prepairing, please wait, this may take some time..."
+echo "Fetching application..."
+if [ -d $PACTFLOW_APPLICATION_DIR ]; then
+    rm -rf $PACTFLOW_APPLICATION_DIR
+fi
+git clone git@github.com:pactflow/pactflow-application.git $PACTFLOW_APPLICATION_DIR >/dev/null 2>&1
+cd $PACTFLOW_APPLICATION_DIR
+echo "Done."
+
+
+####################
+# Find SHA to diff against
+####################
+echo "Pulling latest image..."
+docker pull $ONPREM_PROD_IMAGE >/dev/null 2>&1
+PROD_TAG=$(docker inspect $ONPREM_PROD_IMAGE | jq -r '.[0].ContainerConfig.Env[] | select(startswith("PACTFLOW_GIT_SHA="))' | cut -d "=" -f2)
+
+if [ -z ${DEV_TAG} ]; then 
+  DEV_TAG="HEAD"
+fi
+echo "Done."
+
+
+####################
+# Create branch that will merged and create version in Jira
+####################
+if [ -n "$IS_RELEASE" ]; then  
+  
+  echo "Creating a branch for this release..."
+  cd $DOCS_ROOT_DIT
+  git checkout -b ${RELEASE_VERSION}-release
```

Comment:
Our branch name convention in this repo is `release/<RELEASE_VERSION>`. It doesn't matter too much in this repo, but in other repos, it makes a difference because we can only do certain Buildkite actions on `master` or `release/` and it would be nice to be consistent.